### DISCUSSION TITLE
multiple emits for a second running in background task.

### DISCUSSION THREAD

**Discussion by arun6894** [miguelgrinberg/python-socketio#1005] [General]
I'm trying to run only 2 emits per second to check how many emits the framework can handle for every second in the background task after establishing 1500 client connections. I could see the CPU load gradually increase and see disconnection in a minute.

```
async def task(sid):
    while True:
        await sio.sleep(1.3)
        print("send to all in room ")
        await sio.emit('mult', {'numbers': [3, 4]}, room='b')
        await sio.emit('mult', {'numbers': [3, 4]}, room='b')
```

Is there a better way to check this scenario?

**Comment by miguelgrinberg** [comment]
I don't understand what you want to check. If 1500 client connections produce disconnections, shouldn't you try lowering the  number of connections? 2 emits per second with 1500 clients means you are emitting 3000 messages per second. Can your hardware handle that much traffic?

If the goal is determining emits per second, why don't you use a single connection and emit to it without a sleep? That will give you how many emits per second you can handle.

**Reply by arun6894** [reply]
the goal is to determine emits per second for how many client connections it could as the application works for multiple client connections. So far I could see the single processor could handle 5 emits for 3000 connections using Nginx proxy from 3 different client machines. 1000-1000-1000. Later I ran two processes with the same 5 emits for 6000 connections by 2000-2000-2000. I could see disconnections.  The server machine is running with 2 cores during 3000 connections is it taking 2nd core as well?  When running with 2 processors it couldn't handle? 

With a single processor, it serves 5 emits for 3000 connections. when I double the same emits with 6000 connections I see disconnections. Is it something to do with cores/processors

**Reply by miguelgrinberg** [reply]
You are asking me questions that I cannot answer. This depends on the hardware that you are using, it is not a software matter.

I would imagine two server processes will not double the performance, simply because to go from one CPU to two CPUs you have to add a message queue, which has some overhead.

**Reply by arun6894** [reply]
@miguelgrinberg  I've been using the message queue in server code for the above test run.

```
mgr = socketio.AsyncRedisManager(url=url, channel=channel, write_only=False)
sio = socketio.AsyncServer(async_mode='aiohttp',client_manager=mgr, logger=True, engineio_logger=True)
```
Is there a way to double the process performance of one CPU? I know it's an additional question that can't be answerable. Just want to know your thoughts on it.
